Expensify/App
# [NO QA]Sign commits for @OSBotify

### Details
Second try at signing @OSBotify's commit's for creating new versions and cherry picks. This code was copied from these five reverted PRs:
    1. https://github.com/Expensify/App/pull/6379
    2. https://github.com/Expensify/App/pull/6377
    3. https://github.com/Expensify/App/pull/6374
    4. https://github.com/Expensify/App/pull/6371
    5. https://github.com/Expensify/App/pull/6368

### Fixed Issues

Related https://github.com/Expensify/Expensify/issues/149007 (Doesn't fix 100% yet)

# Tests
1. Merge this PR
2. Verify the version created is signed
3. Verify no other GH actions have issues
